### ISSUE TITLE
Monotone constraints do not work with 'MultiQuantile:alpha=0.05, 0.5, 0.95' loss

### ISSUE THREAD

**Issue by CsanyiDavid** [catboost/catboost#2938]
Similarly to this issue: https://github.com/catboost/catboost/issues/2780#issue-2690873354, I am trying to create a model using monotone constraints and 'MultiQuantile:alpha=0.05, 0.5, 0.95' loss, but monotone constraints did not work. An example:

```python
print(model.get_params())
print(model.predict([28., 87., 54., 69., 65., 67.5, 36.,  1., 0.]))
print(model.predict([100.0, 135.0, 135.0, 135.0, 135.0, 135.0, 50.0, 6.0, 0.0]))
```

```python
{
  'loss_function': 'MultiQuantile:alpha=0.05, 0.5, 0.95',
  'silent': True,
  'monotone_constraints': [1, 1, 1, 1, 1, 1, 1, 1, 0],
  'bagging_temperature': 0.7340280179855684,
  'depth': 10,
  'iterations': 165,
  'l2_leaf_reg': 2.6958276258888714,
  'learning_rate': 0.1522090017408753,
  'random_strength': 4.03950789084095
}
[0.60146347 3.92307589 3.95160885]
[0.0449153  2.63534738 3.38914038]
```

catboost version: 1.2.7
python version: 3.9.23
Operating System: Windows 10 Pro
CPU: Intel Core i7-9750H
